src/main/java/com/amazon/ion/bytecode/bin11/bytearray/ShortTimestampDecoder.kt

```diff
@@ -0,0 +1,221 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.bytecode.bin11.bytearray
+
+import com.amazon.ion.Timestamp
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt16
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt32
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt8AsShort
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedIntAsLong
+import java.math.BigDecimal
+
+/**
+ * Helper class for decoding the various short timestamp encoding variants from a [ByteArray].
+ */
+internal object ShortTimestampDecoder {
```

Comment:
How about just `TimestampDecoder`? There's not really a need to have separate short and long timestamp decoder classes.